app/lib/common/notifications/notifications.dart

```diff
@@ -312,86 +312,191 @@ Future<bool> handleMessage(
-
-    String body = '(new message)';
-    String title = roomDisplayName;
-
-    if (isDm) {
-      final roomMsg = notif.roomMessage();
-      if (roomMsg != null) {
-        final eventItem = roomMsg.eventItem();
-        if (eventItem != null) {
-          final msgContent = eventItem.msgContent();
-          if (msgContent != null) {
-            body = msgContent.body();
-          }
-        }
-      }
-    } else {
-      final roomMsg = notif.roomMessage();
-      if (roomMsg != null) {
-        final eventItem = roomMsg.eventItem();
-        if (eventItem != null) {
-          final msgContent = eventItem.msgContent();
-          if (msgContent != null) {
-            body = msgContent.body();
-          }
-        }
-      }
+    debugPrint('got a notification');
+    await _showNotification(notif);
+    return true;
+  } catch (e) {
+    debugPrint('Parsing Notification failed: $e');
+  }
+  return false;
+}
 
-      final sender = notif.senderDisplayName();
-      body = sender != null ? '$sender: $body' : body;
+Future<ByteArrayAndroidBitmap?> _fetchImage(
+  NotificationItem notification,
+) async {
+  if (notification.hasImage()) {
+    try {
+      final image = await notification.image();
+      return ByteArrayAndroidBitmap(image.asTypedList());
+    } catch (e) {
+      debugPrint('fetching image data failed: $e');
     }
+  }
+  return null;
+}
 
+Future<Person> _makeSenderPerson(NotificationItem notification) async {
+  final sender = notification.sender();
+  if (sender.hasImage()) {
     try {
-      final currentBase =
-          // ignore: use_build_context_synchronously
-          rootNavKey.currentContext!.read(currentRoutingLocation);
-      final isInChat = currentBase == '/chat/${Uri.encodeComponent(roomId)}';
-      debugPrint('current path: $currentBase == /chat/$roomId : $isInChat');
-      if (isInChat) {
-        debugPrint('We are already in the chatroom. Not showing notification.');
-        return false;
-      }
+      final image = await sender.image();
+      return Person(
+        icon: ByteArrayAndroidIcon(image.asTypedList()),
+        key: sender.userId(),
+        name: sender.displayName(),
+      );
     } catch (e) {
-      // ignore this
+      debugPrint('fetching image data failed: $e');
     }
-
-    _showNotification(title, body, roomId, payload);
-    return true;
-  } catch (e) {
-    debugPrint('Parsing Notification failed: $e');
   }
-  return false;
+  return Person(key: sender.userId(), name: sender.displayName());
 }
 
-Future<void> _showNotification(
-  String title,
-  String body,
-  String threadId,
-  String payload,
-) async {
-  const androidNotificationDetails = AndroidNotificationDetails(
+Future<void> _showNotificationOnAndroid(NotificationItem notification) async {
+  String? body;
+  String title = notification.title();
+  String? payload = notification.targetUrl();
+  String? threadId = notification.threadId();
```

Comment:
`threadId` is ... inconsistent in its usage across platforms unfortunately. I started splitting things up into more precise things so we can have a more platform specific usage pattern...

As far as I understand, threadId in iOS/MacOS groups the notifications under one banner - so we do that per roomId. On Android it is honestly unclear to me what `GroupId` (as it is called there) is meant to do prisicely ...

Reply:
> threadId is ... inconsistent in its usage across platforms unfortunately. I started splitting things up into more precise things so we can have a more platform specific usage pattern...

hmm.. Ok, understood.

> On Android it is honestly unclear to me what GroupId (as it is called there) is meant to do prisicely ...

Check below video clip, you will be clear with channel id and group id use-case.
https://www.youtube.com/watch?v=zGIw4MIJn5o

Reply:
are you sure this is the same "group" here? because there are several other features in relating to group in the [AndroidNotificationDetails](https://pub.dev/documentation/flutter_local_notifications/latest/flutter_local_notifications/AndroidNotificationDetails-class.html) that don't really fit the channel-category-group-definition...

Reply:
yeah, reading a little more about it (while latest compiles), I think `groupId` on Android refers to a message-grouping thing: https://developer.android.com/develop/ui/views/notifications/group